### PR TITLE
Use service account for VPN containers

### PR DESCRIPTION
**Pull request by csrwng**
Removes the need for privileged containers. Creates a new SCC (if not present) that allows the capabilities needed by Hypershift VPN containers.

### CODE REVIEW

File: pkg/installer/install.go [hunk 1]
```diff
@@ -817,37 +826,54 @@ func copyFile(src, dest string) error {
...
 	}
-	userSet := sets.NewString()
-	if exists {
-		userSet.Insert(users...)
+	return scc, nil
+}
+
+func ensureVPNSCC(securityClient securityclient.Interface, namespace string) error {
+	scc, err := securityClient.SecurityV1().SecurityContextConstraints().Get("hypershift-vpn", metav1.GetOptions{})
+	if err != nil && errors.IsNotFound(err) {
```
Comment (joelddiaz) [inline pkg/installer/install.go]: if !IsNotFound()

File: pkg/installer/install.go [hunk 2]
```diff
@@ -817,37 +826,54 @@ func copyFile(src, dest string) error {
...
+		if err != nil {
+			return err
+		}
 	}
-	userSet.Insert(svcAccount)
-
-	if err = unstructured.SetNestedStringSlice(obj.Object, userSet.List(), "users"); err != nil {
-		return err
+	userSet := sets.NewString(scc.Users...)
+	svcAccount := fmt.Sprintf("system:serviceaccount:%s:vpn", namespace)
```
Comment (joelddiaz) [inline pkg/installer/install.go]: maybe constify the 'vpn' name

File: pkg/installer/install.go (hunk 1 above)
Comment (csrwng) [reply pkg/installer/install.go]: ha, I could swear I had fixed that locally, thx!

**Comment by csrwng** [comment]
/hold
@joelddiaz tested this and now it's not working, not sure why. Investigating.

**Comment by csrwng** [comment]
ok testing is done, things should be working ok now

**Review by joelddiaz — COMMENTED** [review]
/lgtm

**Pull request merged**